Create Issue and PR Templates

jellypuno opened the pull request:
Signed-off-by: jellypuno <jessielaine.punongbayan@broadcom.com>

## Addresses:
PR and Issue Templates for our repo

## Details:
This PR includes templates for Bug Report, Enhancement Requests, Guidance Requests and PR Template

## Release Notes:
Milestone: v1.5.1
Changelog: Add PR and Issue Templates

Colin-Stone commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,42 @@
+**Desktop (please complete the following information):**
+
+-   OS:
+-   Visual Studio Code Version:
```
Can we just say VSCode. There is sometimes confusion between Visual Studio and VSCode. Do we really need the VSCode version more than Zowe Explorer version?

The CLI and plugins version may be useful but if SCS is installed that's possibly more important

jellypuno replied:
Done. I've added the change

Colin-Stone commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,21 @@
+
+**Describe alternatives you've considered**
+
+<!-- A clear and concise description of any alternative solutions or features you've considered. -->
```
Add to the comment "Any known work arounds"

jellypuno replied:
Done. I've added the change

Colin-Stone commented on .github/ISSUE_TEMPLATE/guidance_request.md:
```diff
@@ -0,0 +1,16 @@
+about: Ask for guidance, "how to", or other questions
+labels: question
+---
+
```
I would prefer to send the user to the slack channel as we are likely to get more eyes on there

jellypuno replied:
> Not sure we need a guidance request but thank you for adding these templates. Some comments for consideration

I could remove it and add the slack channel in our ReadMe file.

Colin-Stone commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,37 @@
+- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
+- [ ] PR Description is included
+- [ ] `npm run vscode:prepublish` has been executed
+- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
```
Maybe have two checkboxes. One for unit tests and one for integration/system tests in case the PR is being performed by someone without a test system

jellypuno replied:
Done.

lauren-li commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,37 @@
+- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
+- [ ] I have added tests (unit and integration) that prove my fix is effective or that my feature works
+- [ ] I have added necessary documentation (if appropriate)
+- [ ] Any dependent changes have been merged and published in downstream modules
```
- Does "dependent changes" mean "any changes dependent on this PR", or does it mean "any changes this PR depends on"?
- Could we clarify what is meant by "downstream modules" (maybe give some examples if applicable)?

jellypuno replied:
My understanding is the first one "any changes dependent on this PR". I will rewrite it. It is confusing.

jellypuno replied:
I've re-written it

lauren-li commented:
- Do we want to add something in the PR template about manually checking for regressions with existing/related functionality? Maybe something like:
`- [ ] Tested manually and no regressions found`

- It might also be helpful to have a checklist for PR reviewers that includes similar items.

jellypuno commented:
> * Do we want to add something in the PR template about manually checking for regressions with existing/related functionality? Maybe something like:
>   `- [ ] Tested manually and no regressions found`
> * It might also be helpful to have a checklist for PR reviewers that includes similar items.

@lauren-li I've added a PR Review Checklist

lauren-li commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -28,9 +28,29 @@ _Put an `x` in the boxes that apply. You can also fill these out after creating
+- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
+- [ ] PR Description is included
+- [ ] `npm run vscode:prepublish` has been executed
+- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
```
Do we want to say here, "All tests and checks have passed"? (Mostly, I'm just trying to say "All unit and integration tests are passing", in addition to the other checks already listed in parentheses. Maybe that should be a separate list item, though.)

jellypuno replied:
@lauren-li The checks that I was talking about here are the ones that we have right now. Like CodeCov, DCO, Jenkins.

I have on separate checks for unit test and integration tests:

- [ ] Unit Tests has been added
- [ ] Integration Tests has been added
- [ ] There is coverage for the added code

lauren-li replied:
@jellypuno That makes sense. I was just thinking that even though there are list items for adding tests, there aren't any items for having the tests actually pass.

jellypuno replied:
aha. It was my assumption that when I say added, it also means passing. Because of the checks that we have. But I can add the word pass there

tjohnsonBCM commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,57 @@
+- [ ] Integration Tests has been added
+- [ ] There is coverage for the added code
+- [ ] I have tested it manually and there are no regressions found
+- [ ] Overall functionaity works
```
"functionaity" -> "functionality"

tjohnsonBCM commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,57 @@
+
+_Put an `x` in the boxes that apply. If you're unsure about any of them, don't hesitate to ask. We're here to help!_
+
+- [ ] The Checklist above is complete before the PR is opened and ready for review.
```
Is it reasonable for the above checklist to be complete before opening a PR? Some things in the first checklist (DCO and CodeCov checks) won't run in branch builds, so the PR must be opened before someone can know if those checks have passed.

jellypuno replied:
should I remove it?

tjohnsonBCM commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,57 @@
+- [ ] Bugfix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] Documentation Update (if none of the other choices apply)
```
Do we want more generic wording for the last option? Perhaps "Updates to Documentation or Tests", as we have a number of PRs open currently that only touch unit test files.

crawr commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,56 @@
+- [ ] I have added necessary documentation (if appropriate)
+- [ ] Any PR dependencies has been merged and published (if appropriate)
+
+## PR Reviewer Checklist
```
As a PR reviewer, I would prefer to look at just 1 set of checklist. 
I would still go through what the PR author marked as completed but I believe we do not need to duplicate the checklist.
As you mentioned above, it is already a reminder of what PR reviewers will look for before merging.

crawr commented on README.md:
```diff
@@ -20,6 +20,7 @@ The Zowe Explorer extension modernizes the way developers and system administrat
 * For information about how to install the extension from a `VSIX` file and run system tests on the extension, see the [Developer README](https://github.com/zowe/vscode-extension-for-zowe/blob/master/docs/README.md).
 * For information about how to develop for Eclipse Theia, see [Theia README](https://github.com/zowe/vscode-extension-for-zowe/blob/master/docs/README-Theia.md).
 * For information about how to create a VSCode extension for Zowe Explorer (a Zowe Explorer extension) see [VSCode extensions for Zowe Explorer](https://github.com/zowe/vscode-extension-for-zowe/blob/master/docs/README-Extending.md).
+* For questions and general guidance, please visit #zowe-explorer channel on [Slack](https://openmainframeproject.slack.com/)
```
I think this is a good addition, but I am wondering if a slack badge would be better? It would be next to the codecov badge and I believe will make it more visible.

crawr commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,56 @@
+<!-- Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. -->
+
+## Release Notes
+<!-- Include the Milestone Number and a small description of your change that will be added to the changelog -->
```
If there is a linked issue, it should have the same milestone as the PR.

jellypuno commented:
@tjohnsonBCM @lauren-li @crawr @Colin-Stone 

Added based on all your comments. I also combined the checklist for contributor and reviewer. I revised the description so it would be clearer (I hope!). I added a Slack Badge as well. For your review please... Thank you.

tjohnsonBCM commented on README.md:
```diff
@@ -1,6 +1,7 @@
 # Zowe Explorer
 
 [![codecov](https://codecov.io/gh/zowe/vscode-extension-for-zowe/branch/master/graph/badge.svg)](https://codecov.io/gh/zowe/vscode-extension-for-zowe)
+[![slack](https://img.shields.io/badge/chat-on%20Slack-blue?style=flat-square)](https://slack.openmainframeproject.org/)
```
The style of the badges doesn't match:
![image](https://user-images.githubusercontent.com/50887074/81176352-0c624800-8f73-11ea-8505-1cdf3ca7faa5.png)

If `?style=flat-square` is removed from the Slack badge URL, then both badges should have rounded corners.

jellypuno replied:
Done.
